shared/cryptography/codeql/cryptography/Model.qll

```diff
@@ -2178,7 +2222,14 @@ module CryptographyBase<LocationSig Location, InputSig<Location> Input> {
     /**
      * Gets the key size variant of this algorithm in bits, e.g., 128 for "AES-128".
      */
-    string getKeySizeFixed() { result = instance.asAlg().getKeySizeFixed() } // TODO: key sizes for known algorithms
+    string getKeySizeFixed() {
+      result = instance.asAlg().getKeySizeFixed()
+      or
+      exists(int size |
```

Comment:
This implementation points to inconsistent logic; either we put the responsibility of defining the key size on the modeller in all instances or we accept that we need to "fill in the blank with an implicit default" in other cases.

This needs to be defined, with one clear path and instructions added as QLDoc comments on the instance level for modellers. If that's a global predicate in Model.qll that provides defaults, so be it. Otherwise, we need to define which types need to have a size provided and which don't.

Reply:
Also, we should unify `getDigestLength` and `getKeySizeFixed` as well as the `Length`/`Size`/`Fixed`/ terminology.